### PR TITLE
feat(map): Add map constraints and enhance UI settings configuration

### PR DESCRIPTION
**Pull request by TonyGnk**
## Overview
This PR introduces map constraint configurations and fixes property copying in UI settings. It enhances the map's customization capabilities while maintaining proper state management.

### 1. Add map constraints and bounds configuration
- Added new properties to `MapProperties`:
  - `minZoom`, `maxZoom`: Control zoom level constraints
  - `minPitch`, `maxPitch`: Control pitch angle constraints
  - `latLngBounds`: Restrict map viewport to specific bounds
- Added corresponding property application in `MapLibreMap.applyProperties()`
- Implemented proper `copy()` function for immutability
- Updated `equals()` and `hashCode()` to handle new properties

### 2. Enable workflow_dispatch for manual workflow runs
- Added manual trigger capability for workflows

### 3. Fix UiSettings property copying
- Updated secondary constructor to properly copy all properties
- Expanded from single property copy to comprehensive property copying

## Potential Issues and Bugs Found

1. **MapProperties Class**
   - **Bug Risk**: The `var` to `val` change for `maxZoom` is a breaking change that might affect existing code.
   - **Mitigation**: Need migration guide for users relying on mutable `maxZoom`.

2. **Range Validation**
   - **Potential Issue**: No runtime validation for min/max relationships.
   - **Example**: `minZoom` could be set higher than `maxZoom`.
   - **Recommendation**: Add validation in constructor.

### CODE REVIEW

**Comment by klemensz** [comment]
Oh... looks like I forgot to update the secondary constructor in my previous PR 😆

**Review by JonasVautherin — APPROVED** [review]
Looks good, thanks a lot! Let's see what the CI says 👀 

EDIT: @TonyGnk: looks like I merged a conflicting PR. Would you mind rebasing and fixing the conflicts? 🙏

**Comment by TonyGnk** [comment]
Thanks for the review! I'll rebase now,

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
     - 'main'
+  workflow_dispatch:
```
Comment (JonasVautherin) [inline .github/workflows/main.yml:7]: I don't think we need this, do we? 🤔

**Review by JonasVautherin — COMMENTED** [review]
Looks good! Just wondering about the `workflow_dispatch`, but the rest is fine for me 🚀

File: .github/workflows/main.yml (hunk 1 above)
Comment (TonyGnk) [reply .github/workflows/main.yml:7]: I initially added workflow_dispatch for easier testing in my fork, but you're right - it's not really needed in the main repo. I'll remove it!

**Review by JonasVautherin — APPROVED** [review]
Thank you so much! 🚀

**Pull request merged**